internal/driver/controller.go

```diff
@@ -66,12 +84,38 @@ func (s *ControllerService) CreateVolume(ctx context.Context, req *proto.CreateV
 		location = *loc
 	}
 
+	var volumeLabels = map[string]string{
+		tagKeyCreatedBy: "csi-driver",
+	}
+
+	maps.Copy(volumeLabels, s.extraVolumeLabels)
+
+	for key, value := range req.GetParameters() {
+		switch strings.ToLower(key) {
+		case parameterKeyPVCName:
+			volumeLabels[tagKeyCreatedForClaimName] = value
+		case parameterKeyPVCNamespace:
+			volumeLabels[tagKeyCreatedForClaimNamespace] = value
+		case parameterKeyPVName:
+			volumeLabels[tagKeyCreatedForVolumeName] = value
+		case parameterKeyLabels:
+			customLabels, err := utils.ConvertLabelsToMap(value)
+			if err != nil {
+				return nil, status.Errorf(codes.InvalidArgument, "Invalid format of parameter labels: %s", err)
+			}
+			maps.Copy(volumeLabels, customLabels)
+		default:
+			return nil, status.Errorf(codes.InvalidArgument, "Invalid parameter key %s for CreateVolume", key)
+		}
```

Comment:
We previously did not validate that the parameters are from a known list, so adding this now would be a breaking change. Please ignore unknown parameters instead.